processor/k8sattributesprocessor/internal/kube/client.go

```diff
@@ -454,41 +454,104 @@ func (c *WatchClient) addOrUpdatePod(pod *api_v1.Pod) {
 		}
 	}
 
+	return newPod
+}
+
+// getIdentifiersFromAssoc returns list of PodIdentifiers for given pod
+func (c *WatchClient) getIdentifiersFromAssoc(pod *Pod) []PodIdentifier {
+	if pod.Attributes == nil {
+		pod.Attributes = map[string]string{}
+	}
```

Comment:
I don't this part is needed. Let's avoid unnecessary allocations

Reply:
removed